app/src/main/java/com/example/ankas/CategoryAndProduct.java

```diff
@@ -0,0 +1,242 @@
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.PopupMenu;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.example.ankas.Adapter.CategoryAdapter;
+import com.example.ankas.Adapter.ProductAdapter;
+import com.example.ankas.Components.ExpandableHeightGridView;
+import com.example.ankas.Fragments.MainFragment;
+import com.example.ankas.Objects.Category;
+import com.example.ankas.Objects.Product;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.w3c.dom.Text;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class CategoryAndProduct extends AppCompatActivity {
+    List<Category> categoryList;
+    List<Product> productList;
+    ExpandableHeightGridView grid_categoryAndProduct;
+    // Название
+    TextView txt_title;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_category_and_product);
+        // Верхнее меню
+        toolbar();
+        grid_categoryAndProduct = findViewById(R.id.grid_categoryAndProduct);
+        txt_title = findViewById(R.id.txt_title);
+        int id_ = getIntent().getIntExtra("id_", 0);
+        categoryList = new ArrayList<>();
+        productList = new ArrayList<>();
+        new getCategoryAndProduct().execute("http://anndroidankas.h1n.ru/mobile-api/Product/ProductOrCategory/" + id_);
+    }
+    // Верхнее меню
+    private void toolbar() {
+        ImageView img_logo = findViewById(R.id.img_logo);
+        ImageView img_search = findViewById(R.id.img_search);
+        img_logo.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent intent = new Intent(CategoryAndProduct.this, MainActivity.class);
+                intent.putExtra("ItemFragment", R.id.item_main);
+                startActivity(intent);
+            }
+        });
+        img_search.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent intent = new Intent(CategoryAndProduct.this, SearchActivity.class);
+                startActivity(intent);
+            }
+        });
+    }
+
+    // Получение товаров или категорий
+    private class getCategoryAndProduct extends AsyncTask<String, Void, String> {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Class names should start with an upper case letter (NM_CLASS_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-class-names-should-start-with-an-upper-case-letter-nm-class-naming-convention).
Class names should be nouns, in mixed case with the first letter of each internal word capitalized. Try to keep your class names simple and descriptive. Use whole words-avoid acronyms and abbreviations (unless the abbreviation is much more widely used than the long form, such as URL or HTML).